qiskit/optimization/algorithms/grover_optimizer.py

```diff
@@ -294,9 +288,8 @@ def _get_probs(self, qc: QuantumCircuit) -> Dict[str, float]:
             shots = self.quantum_instance.run_config.shots
             hist = {}
             for key in state:
-                hist[key] = state[key] / shots
+                hist[key[::-1]] = state[key] / shots
```

Comment:
Why did this need to change? Was it wrong before or has the order been reversed somewhere?

Reply:
It was wrong before. The bit order is inverted for the state vector simulator to be the conventional bit order (e.g. the leftmost bit is the highest bit).`keys = [bin(i)[2::].rjust(int(np.log2(len(state))), '0')[::-1]
                    for i in range(0, len(state))]`
And, the grover optimizer works with the conventional bit order (it uses bitwise operations such as bitwise adding with the conventional bit order). 

I guess no-one tested before since measurement was not added and it simply didn't move before.